node.cpp

```diff
@@ -0,0 +1,49 @@
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include "node.h"
+
+#define MAX_CHILDREN 100
+
+Node *node_new(const char *name)
+{
+  Node *node = (Node *)calloc(1, sizeof(Node));
+  node->internal_type = name;
+  return node;
+}
+
+void node_add_role(Node *node, uint16_t)
+{
+}
+
+void node_add_child(Node *parent, Node *child)
+{
+  if (!parent || !child)
+  {
+    return;
+  }
+  if (!parent->children)
```

Comment:
Node constructor should probably guarantee that a node has an empty children vector at construction time